walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -36,45 +38,77 @@
     }
     
     // Filtering out events with no conflicting attendees.
-    for(Event currentEvent:events) {
-      Collection<String> eventAttendees = currentEvent.getAttendees();
-      for(String currentAttendee : attendees) {
-        if(eventAttendees.contains(currentAttendee)) {
-          conflictEvents.add(currentEvent.getWhen());
-          break;
-        }
-      }
-    }
-    
-    // If no events conflict then the whole day is free.
+    TreeSet<TimeRange> conflictEvents = getConflicts(events, attendees);
+     
+    // If no events conflict then the whole day is free. Otherwise find
+    // available blocks of time.
     if(conflictEvents.isEmpty()){
-      availableTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true));
-      return new ArrayList<TimeRange>(availableTimes);
+      availableTimes.add(TimeRange.WHOLE_DAY);
```

Comment:
don't you want to return early still if there's no conflicts?

Reply:
These conflicts are only for the mandatory attendees and in the case where there are no mandatory attendees conflictEvents.isEmpty() is true but it could still be the case there are optional attendees so their availability needs to be taken into account before returning. Sorry for the confusion, I should've made it clear these are conflicts only for the mandatory attendees.